### PR TITLE
Dry-run: cache cloned repos

### PR DESCRIPTION
**Pull request by feelepxyz**
Cache the cloned repo when `--cache=files` is passed to the dry-run
script.

This also allows setting up a test repo in `tmp/repo/name` and passing
`--clone` and `--cache=files` to load it for any package manager.

The codes in the dry-run scripts are not pretty but punting on a wider
refactor as this probably means re-writing it.

### CODE REVIEW

File: bin/dry-run.rb [hunk 1]
```diff
@@ -476,8 +475,20 @@ def handle_dependabot_error(error:, dependency:)
...
 fetcher = Dependabot::FileFetchers.for_package_manager($package_manager).new(**fetcher_args)
-$files = if $options[:clone] || always_clone
+$files = if $repo_contents_path
+           if $options[:cache_steps].include?("files") && Dir.exist?($repo_contents_path)
+            puts "=> reading cloned repo from #{$repo_contents_path}"
+           else
+            puts "=> cloning into #{$repo_contents_path}"
+            FileUtils.rm_rf($repo_contents_path)
+           end
            fetcher.clone_repo_contents
```
Comment (feelepxyz) [inline bin/dry-run.rb:485]: FYI: This method short circuits if the repo exists https://github.com/dependabot/dependabot-core/blob/40baefd1e812feca9712daa1e359583a29b5fdc1/common/lib/dependabot/file_fetchers/base.rb#L477

File: bin/dry-run.rb [hunk 2]
```diff
@@ -476,8 +475,20 @@ def handle_dependabot_error(error:, dependency:)
...
+           if $options[:commit]
```
Comment (feelepxyz) [inline bin/dry-run.rb:486]: Note: the branch is handled in `clone_repo_contents` but `commit` is a dry-run only option so handled here

**Review by jurre — APPROVED** [review]
My only concern is that bin/dry-run.rb keeps getting bigger and bigger and it's pretty hard to grasp what's going on

**Comment by feelepxyz** [comment]
> My only concern is that bin/dry-run.rb keeps getting bigger and bigger and it's pretty hard to grasp what's going on

Yup good call out, would be worth trying to do a series of smaller refactors to split it up. Will see if I can find a good starting point.

**Pull request merged**